WChat/src/app/redux/group/reducer.js

```diff
@@ -23,19 +29,43 @@ const group = (state = initialState, action) => {
         groupsLoading: false,
         groupsError: action.error
       });
-    case actionTypes.GET_GROUP_CHAT_SUCCESS:
-      const oldGroupIndex = state.groups.findIndex((group) => group.id === action.groupId);
-      const oldGroup = state.groups[oldGroupIndex];
-      let newGroup =  {
+    case actionTypes.GET_GROUP_CHAT_SUCCESS:      
+      if(!action.chats.length){
+        return Immutable.merge(state, {
+          currentGroupChat: []
+        })
+      }
+      oldGroupIndex = state.groups.findIndex((group) => group.id === action.groupId);
+      oldGroup = state.groups[oldGroupIndex];
+      newGroup =  {
         ...oldGroup,
         chats: action.chats,
         lastChat: action.chats.reduce((latest, current) => latest.createdAt > current.createdAt ? latest : current)
       }
 
+      newGroups = cloneAndInsertInArray(state.groups, newGroup, oldGroupIndex)
+      return Immutable.merge(state, {
```

Comment:
instead of calling Immutable.merge, wrap the `initialState`of the reducer with `Immutable`, so all the state is immutable. And you can call `state.merge({ ... })`